### DISCUSSION TITLE
Disable tilt restore for venetian blinds

### DISCUSSION THREAD

**Discussion by s0170071** [arendst/Tasmota#14072] [Q&A]
I am running Tasmota 10.1.0 on a shelly25. I control my venetian blinds with openhab->mqtt->shelly.
Tilt control is set up and works as [described](https://tasmota.github.io/docs/Blinds-and-Shutters/), but I found I would rather not have the tilt restored when I move the blind to a different position. 
Is there any way to switch tilt restore off ?

**Comment by steiny2k** [comment]
Hi @stefanbode, I would also be interested in such a behaviour. This correction of the tilting is not quite intuitive, users are used to control the blinds with the switches in the wall.

**Reply by stefanbode** [reply]
We can bring this to a more broad discussion. If you change shutterposition what do you expect? Most users like the restore of the tilt. I would not like to change this because this is one of the benefits of the tilt add on. If you do not like the restore, why do you define the tilt if operating with buttons anyhow. In this case I would advise to remove tilt definition completely. I already changed end position tilt behavior to ensure it stay where it is. Please take it not as critic. I do want to understand your use case.

**Reply by s0170071** [reply]
**TLDR;** 
blind is tilt, wife wants it closed. Tasmota  restores the tilt instead. This endangers me and my whole automation playground !

**Long:**
If memory serves well, I did use the tilt function only because the shutter position can just be set in integer steps. I filed an issue about this but it was just swiftly closed -  as usual. If you could use finer granularity, then you can set the tilt yourself by just moving the blind. 

Here is what I am used from my  Homematic devices. 
- You press up, it moves up. You press up or down - it stops. (Tasmota moves it down on first down press) 
- You press up and up again quickly, it stops in a tilted position. 
- you move the blind remotely  from 100% to 99.5% it tilts 45°. The blind tilt time is about 1 sec, the bottom to top run time is 120sec. What I need is 0.5 sec run time which is less than 0.5% 

My use case: when the blind is closed, I want it to tilt 45° when I open the window. But: if I then move it manually in any direction, I like it to stay. I would love an option to a) tilt once without restore or b) cancel the tilt position without moving the blind
Oh- and the cherry on the cake would be the mentioned above "either button stops the movement" function instead of going straight to the opposite direction.

**Reply by steiny2k** [reply]
Hi Stefan,

I love the TLDR by @s0170071! Similar situation for me. But I took it back to the customer (my wife) and we agreed to leave tilt-restore for now and see if it works for her.

I setup the tilt function in Tasmota because I like to be able to set the tilt with automatons out from Home Asssistant. But my expectation was the overall user experience wouldn't change.
My setup are "Raffstoren" (see https://www.warema.com/de-de/jalousien/raffstore.html) controlled by a simple up-/down switch, not buttons (see https://hager.com/de-ch/katalog/produkt/wh25191300c-kaline-storenschalter).

Thanks a lot for the hard work and contributions! Much appreciated!

**Reply by stefanbode** [reply]
Regarding the request of using "99.5%". This was rejected because a major refactoring is required. All the commands normally only just pass integer. Not real variables. It is possible to add string evaluation and multiply by 10. Additionally the position and all similar variables are int_8 and therefore limit to 128 and not 1024 what would be needed. Then there was an additional reason to reject this: The minimum time slice is 0.05 seconds and therefore the minimum runtime about 0.25 seconds or more. Therefore very tiny changes are complicated anyhow from the underlying basics. And this is the DNA of Tasmota we cannot change at all.

With reasonable amount we can ad a config option to restore the tilt yes/no. This is something we can negotiate about. Default would be current behaviour and you can disable the restore. I assume you would like this to get implemented

**Reply by stefanbode** [reply]
Regarding: 
"You press up, it moves up. You press up or down - it stops. (Tasmota moves it down on first down press)"

This is something that already is implemented when you use the shutterbuttons functionality (as far as I can see, shutterstop send on any button if moving is detected) . But be aware that this is only for buttons. If you switch the relay directly, this will not work because the SWITCH permanently put power on the relay2. Same if you use buttons directly to set the relay. In this case you power on directly relay2 and the driver can only react but not change the behavior.

**Reply by stefanbode** [reply]
There is a recent change already published to DEVELOPMENT that if your target is 0 or 100 the tiltposition is reset. This means, if you use the shutterbuttons then the OPEN will automatically kill the tiltposition and the shutter should stop and NOT restore the tilt anymore. This all required that you do not DIRECTLY set the relay to ON. 

line 1260: it does not matter if you reach 0 or 100. Just the intend to go there deletes the tiltposition.
```
  // if position is either 0 or 100 reset the tilt to avoid tilt moving at the end
      if (XdrvMailbox.payload ==   0 && ShutterRealToPercentPosition(Shutter[index].real_position, index)  > 0  ) {Shutter[index].tilt_target_pos = Shutter[index].tilt_config[4];}
      if (XdrvMailbox.payload == 100 && ShutterRealToPercentPosition(Shutter[index].real_position, index)  < 100) {Shutter[index].tilt_target_pos = Shutter[index].tilt_config[3];}

```

**Reply by ma-zal** [reply]
I am also searching for how to disable the tilt restoration after a position change. This is because using physical buttons looks non-intuitive (different behavior than the non-smart controlled solutions).

My idea is that a new SetOption could be very helpful. Something like:

> `SetOptionXX` - Disable shutter tilt position restore after position change.

I feel the solution as good because the current behavior can stay untouched for this case. And the automatic restoration could be suppressed by setting the `SetOptionXX 1` by anybody based on the desired behavior.

The potential implementation looks relatively simple: To extend the mentioned `if` conditions for end positions `0` and `100` with the new rule: `or SetOptionXX is ON`. [Mentioned original code (candidate, where to implement this new feature)](https://github.com/stefanbode/Sonoff-Tasmota/blob/35887f6edec07a22d615fbc82dc5d482e63f1e71/tasmota/tasmota_xdrv_driver/xdrv_27_shutter.ino#L1334-L1335).

**Reply by stefanbode** [reply]
You can use shutterbuttons and define the tilt with the pressed button. This should bring you the requested behavior. Please see documentation how to set the tilt with shutterbuttons. Works only on esp32.

**Reply by ma-zal** [reply]
> @stefanbode wrote: 
> You can use shutterbuttons ... Works only on esp32.

Mentioned limitation to esp32 is crucial. Many devices run on ESP8266 actually, so this feature is not available for them. :-(

**Reply by stefanbode** [reply]
Yes. Might be. Shelly Pro devices are cheap and really reliable. The do work with ESP32 and are easy to flash with Tasmota. My recommendation for complex shutters.

**Reply by ma-zal** [reply]
Your recommendation to replace a device with ESP32 is not realizable in many cases, @stefanbode. People have specific devices like wall switches, which have specific dimensions, specific designs, etc. So, unfortunately, it is impossible to easily find an identical alternative with ESP32.

By the way, this is a software development discussion about Tasmota firmware, so I personally feel that solutions like "hardware replacement" or "switching to Shelly" are a bit off-topic. An eventual decision "will not be implemented" can be absolutely valid and that's enough.

**Reply by stefanbode** [reply]
You get my Shelly answer wrong. I was suggesting to buy a Shelly and flash it with Tasmota. The new Shelly has esp32 and offers these options. If you do not want to restore the tilt, isn't it easier just NOT to configure the tilt. Then you got your requested behavior.

**Comment by s0170071** [comment]
After I did not get the results I wanted with Tasmota I tried the shelly 
stock firmware.  Worked right out of the box....


Am 2. Juli 2024 19:22:44 schrieb stefanbode ***@***.***>:

**Comment by spoler** [comment]
I had a same issue and have managed to come up with workaround. 

My tilt configuration is following:
`ShutterTiltConfig1 -90 90 58 -90 0`

I am using following rules:
_#Saving Target and Tilt position in variable on Target/Tilt change_
`Rule1 ON Shutter1#Target!=%Var1% DO Var1 %value% ENDON ON Shutter1#Tilt!=%Var2% DO Var2 %value% ENDON `

_#When Tilt reaches it's end position (90 and -90 in my case) send new shutterposition command with existing target and end position that was reached_
`Rule2 ON Var2#state==90 DO shutterposition %var1%,%var2% ENDON ON Var2#state==-90 DO shutterposition %var1%,%var2% endon`

Tilt is not restored and it remains in the end position that was reached during the movement. 
Hope this helps.

**Reply by stefanbode** [reply]
As more and more people are asking for a disable option of the tilt restore I will check for it on esp32. At the end position 0 and 100 the tilt is not restored anyhow. It is only if you move from 10 to 50 ist restores the tilt that was set before.
